### ISSUE TITLE
the reconstruct performance of Learning to Reconstruct

### ISSUE THREAD

**Issue by zyj008** [zhedongzheng/finch#7]
Hello. You used VAE for reconstruct sentences from imdb. According to your results, I think the reconstruction performance is not good. The reconstructed sentences have great difference with original ones. I am new in NLP, so I want to know can complete reconstruction be achieved with existing models. Can you give me some hints about the reasons which caused bad reconstruction performance? Is it due to the simple model or lack of training or something else? Thank you.

**Comment by zhedongzheng** [comment]
Hi, VAE for NLP is much harder to train than that for Computer Vision.
The complete reconstruction is too hard to achieve, but we aim for generating sentence with similar meanings with different words (paraphrase).
The hints for NLP are:
1. apply word dropout (on decoder inputs): this forces decoder to rely on the encoder message
2. concat z to decoder inputs: this helps decoder get access to z in any timestep
The [results](https://nbviewer.jupyter.org/github/zhedongzheng/finch/blob/master/src_nlp/tensorflow/vae/train.ipynb) now get better after adding these two tricks, and these two tricks are mentioned in the paper "Generating sentence from a continuous space"

**Comment by zhedongzheng** [comment]
TF2 will replace TF1 in the near future, therefore I have to rewrite all the code.
Previous issues will be closed. Sorry for any inconvenience.
